### PR TITLE
Edited documentation and removed SMTP configs from env

### LINKED ISSUE #2177: Update SMTP Config documentation and remove old config from env template
With Autolab 3.0 changes, SMTP Mailing configurations moved to the config interface, but documentation / env files still reference the old setup method.

There should be updates to the documentation on SMTP config setup using the interface, removal of smtp related `.env` variables in `.env.template`, and removal of smtp related variables in `production.rb.template`

### PR DESCRIPTION
**Pull request by KesterTan**
## Description
This resolves #2177. With the recent change, SMTP configs are no longer taken from the env file but are configured using the UI and saved in the config. For more information see #2168. This PR updates the documentation with instructions to use the UI to configure SMTP and for application and production.rb files to reference the config instead.

## How Has This Been Tested?
- Test manual installation and docker installation of Autolab with these changes, check that you can save configuration and when you send an email, it works and its received which shows that SMTP configuration is a success. 

Screenshot 2024-09-02 at 11 50 00 PM
Screenshot 2024-09-02 at 11 50 09 PM

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have run rubocop and erblint for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [x] My change requires a change to the documentation, which is located at [Autolab Docs](https://docs.autolabproject.com/)
- [x] I have updated the documentation accordingly, included in this PR

### CODE REVIEW

File: config/environments/production.rb.template [hunk 1]
```diff
@@ -76,7 +76,7 @@ Autolab3::Application.configure do
   # config.active_record.auto_explain_threshold_in_seconds = 0.5
 
   # default from 
-  ActionMailer::Base.default :from => ENV.fetch('SMTP_DEFAULT_FROM','something@example.com')
```
Comment (20wildmanj) [inline config/environments/production.rb.template:79]: I don't think config hash is defined here? config_hash is only non-nil if there exists the yml file, see line 113

File: config/application.rb [hunk 2]
```diff
@@ -115,7 +115,7 @@ class Application < Rails::Application
     config.site_version = "3.0.0"
 
     # Set application host for mailer
-    config.action_mailer.default_url_options = { host: ENV['MAILER_HOST'] || "YOUR_APP_HOST" }
```
Comment (20wildmanj) [inline config/application.rb:118]: I guess a weird thing here is that this option doesn't really get reflected on the config side, but I did see damian's comment so I guess just removing the env part is fine?

File: docs/installation/mailing.md [hunk 3]
```diff
@@ -2,10 +2,18 @@
 
 Autolab requires mailing to allow users to register accounts and reset passwords. You will also be able to make announcements through Autolab as well. The recommended approach is to setup Autolab to use a SMTP server, such as [mailgun](https://mailgun.com), [SendGrid](https://sendgrid.com), [Amazon SES](https://aws.amazon.com/ses/) or any other valid SMTP mail servers to send out email.
 
-We intend this instructions mainly for production usage.
```
Comment (20wildmanj) [inline docs/installation/mailing.md:5]: You may want to add a note / footnote about how now we only use the UI config and that env file based config has been removed as a disclaimer.

File: docs/installation/mailing.md [hunk 4]
```diff
@@ -2,10 +2,18 @@
 
 Autolab requires mailing to allow users to register accounts and reset passwords. You will also be able to make announcements through Autolab as well. The recommended approach is to setup Autolab to use a SMTP server, such as [mailgun](https://mailgun.com), [SendGrid](https://sendgrid.com), [Amazon SES](https://aws.amazon.com/ses/) or any other valid SMTP mail servers to send out email.
 
-We intend this instructions mainly for production usage. 
+We intend this instructions mainly for production usage.
```
[the remark this answers is not in the document]
Comment (20wildmanj) [reply docs/installation/mailing.md:5]: It is indeed a typo

File: docs/installation/mailing.md (hunk 3 above)
Comment (20wildmanj) [reply docs/installation/mailing.md:5]: Oh okay I see that all of the old instructions are preserved, but maybe make a heading for v3.0.0 so it's clear that for newer versions they should use the UI config and after and add a screenshot or something of the page?

**Review by 20wildmanj — APPROVED** [review]
At the very least, I don't think anything breaks from the removal of the env variables. I was able to get SMTP service up and running on my end, although we should get it set up on nightly as well so we can test it over there as well. Just some more clarity in documentation / check nit for production.rb.template

File: config/environments/production.rb.template (hunk 1 above)
Comment (KesterTan) [reply config/environments/production.rb.template:79]: Moved it down

Comment (20wildmanj) [reply config/environments/production.rb.template:79]: I don't see this change on github?

Comment (KesterTan) [reply config/environments/production.rb.template:79]: Should be updated now.

File: docs/installation/mailing.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (KesterTan) [reply docs/installation/mailing.md]: Fixed it.

File: docs/installation/mailing.md (hunk 3 above)
Comment (KesterTan) [reply docs/installation/mailing.md:5]: Made a heading.

**Pull request merged**